### PR TITLE
Fix oom error

### PR DESCRIPTION
**Pull request by KatalinaBobowik**
Fixing OOM error due to serializing data frames. This now runs successfully in the `test` bucket.

### CODE REVIEW

File: scripts/eqtl_hail_batch/conditional_analysis.py [hunk 1]
```diff
@@ -77,17 +77,20 @@ def get_genotype_df(residual_df, gene_snp_test_df):
     # only keep samples that are contained within the residuals df
     # this is important, since not all indivuduals have expression/residual
     # data (this varies by cell type)
-    samples_to_keep = set(residual_df.sampleid)
-    set_to_keep = hl.literal(samples_to_keep)
-    mt = mt.filter_cols(set_to_keep.contains(mt['onek1k_id']))
+    samples_to_keep = hl.literal(list(residual_df.sampleid))
```
Comment (lgruen) [inline scripts/eqtl_hail_batch/conditional_analysis.py]: `contains` is likely faster on a `set` compared to a list. Not sure if you need to first convert to a list (`hl.set(list(residual_df.sampleid))`) though.

```suggestion
    samples_to_keep = hl.set(residual_df.sampleid)
```

File: scripts/eqtl_hail_batch/conditional_analysis.py [hunk 2]
```diff
@@ -316,43 +330,37 @@ def spearman_correlation(df):
...
+            x[['chrom', 'bp', 'a1', 'a2', 'gene_symbol', 'cell_type_id', 'round']]
+        ),
+        axis=1,
+    )
+    adjusted_spearman_df['variant_id'] = adjusted_spearman_df.apply(
+        lambda x: ':'.join(x[['chrom', 'bp', 'a2']]), axis=1
+    )
+    adjusted_spearman_df['snp_id'] = adjusted_spearman_df.apply(
+        lambda x: ':'.join(x[['chrom', 'bp', 'a1', 'a2']]), axis=1
+    )
```
Comment (lgruen) [inline scripts/eqtl_hail_batch/conditional_analysis.py]: Didn't we want to remove those IDs?

**Review by lgruen — CHANGES_REQUESTED** [review]
Did you already test this with the `v2` test inputs?

File: scripts/eqtl_hail_batch/generate_eqtl_spearman.py [hunk 3]
```diff
@@ -656,7 +661,8 @@ def main(
     copy_common_env(filter_mt_job)
     filtered_mt_path = filter_mt_job.call(prepare_genotype_info, keys_path=keys)
 
-    for idx in range(n_genes_in_scatter):
+    # for idx in range(n_genes_in_scatter):
```
Comment (illusional) [inline scripts/eqtl_hail_batch/generate_eqtl_spearman.py]: Should this be reverted?

Comment (KatalinaBobowik) [reply scripts/eqtl_hail_batch/generate_eqtl_spearman.py]: yes, good catch, thanks! I'll update that

File: scripts/eqtl_hail_batch/conditional_analysis.py (hunk 2 above)
Comment (KatalinaBobowik) [reply scripts/eqtl_hail_batch/conditional_analysis.py]: Sorry yes, I forgot about those. That's removed now

File: scripts/eqtl_hail_batch/conditional_analysis.py (hunk 1 above)
Comment (KatalinaBobowik) [reply scripts/eqtl_hail_batch/conditional_analysis.py]: `hl.set(list(residual_df.sampleid))` works 👌

**Review by lgruen — APPROVED** [review]

**Pull request merged**